### PR TITLE
#4101-UI-Implement CV version removal

### PR DESCRIPTION
**Pull request by ldjebran**
```console
(2.7) dlezz@elysion:~/projects/robottelo-fork$ py.test tests/foreman/ui/test_contentview.py -k "test_positive_remove_cv_version_from_default_env or test_positive_remove_renamed_cv_version_from_default_env or test_positive_remove_promoted_cv_version_from_default_env or test_positive_remove_qe_promoted_cv_version_from_default_env or test_positive_remove_prod_promoted_cv_version_from_default_env or test_positive_remove_cv_version_from_env or test_positive_remove_cv_version_from_multi_env or test_positive_delete_cv_promoted_to_multi_env"
================================================= test session starts ==================================================
platform linux2 -- Python 2.7.12, pytest-3.0.3, py-1.4.31, pluggy-0.4.0
rootdir: /home/dlezz/projects/robottelo-fork, inifile: 
plugins: cov-2.4.0, xdist-1.15.0
collected 81 items 

tests/foreman/ui/test_contentview.py ...s.s....

================================================= 71 tests deselected ==================================================
================================ 8 passed, 2 skipped, 71 deselected in 2459.14 seconds =================================
(2.7) dlezz@elysion:~/projects/robottelo-fork$ 
```
issue: ##4101

### CODE REVIEW

File: robottelo/ui/contentviews.py [hunk 1]
```diff
@@ -614,3 +614,45 @@ def package_search(self, name, version_name, package_name,
...
+        return self.wait_until_element(
+            locators.contentviews.version.puppet_module_name % module_name)
+
+    def remove_version_from_environments(self, name, version, environments):
+        """Remove a content view version from lifecycle environments"""
+        # find and open the content view
+        self.search_and_click(name)
+        # click on the version remove button
+        self.click(locators.contentviews.remove_ver % version)
+        # ensure, that remove Completely remove version check box is unchecked
```
Comment (abalakh) [inline robottelo/ui/contentviews.py]: nitpick: please fix the comment

File: tests/foreman/ui/test_contentview.py [hunk 2]
```diff
@@ -33,8 +33,10 @@
     CUST_PERMS_LIST,
     DEFAULT_CV,
     DOCKER_REGISTRY_HUB,
+    DOCKER_UPSTREAM_NAME,
     ENVIRONMENT,
     FAKE_0_PUPPET_REPO,
+    FAKE_0_PUPPET_MODULE,
```
Comment (abalakh) [inline tests/foreman/ui/test_contentview.py]: nitpick: please follow alphabetical order

File: tests/foreman/ui/test_contentview.py [hunk 3]
```diff
@@ -4112,10 +4521,84 @@ def test_positive_delete_cv_promoted_to_multi_env(self):
...
+            )
+            # promote the content view to all lifecycle environments
+            for env_name in env_names:
+                self.content_views.promote(cv_name, version, env_name)
+            self.assertEqual(
+                set(self._get_cv_version_environments(version)),
+                all_env_names_set
+            )
+            # remove all the environments from version
+            self.content_views.remove_version_from_environments(
```
Comment (abalakh) [inline tests/foreman/ui/test_contentview.py]: The docstring of current test is definitely outdated and you've implemented another test.
You should either fix the according to your implementation and/or try to delete cv promoted to some lces and ensure error message was risen.

Comment (abalakh) [inline tests/foreman/ui/test_contentview.py]: The best option would probably be to split this test into 2 separate ones - 1 which you currently implemented (but with updated docstring), another one (negative) which tries to delete the cv which has lces associated and ensures error message is there and cv was not deleted.

Comment (ldjebran) [inline tests/foreman/ui/test_contentview.py]: yes you are right, here the cv cannot be removed when promoted to lcves, need a negative test of this

**Pull request merged**